api/api.go

```diff
@@ -0,0 +1,45 @@
+package api
+
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/date"
+)
+
+// AzureProvider is an interface to access underlying Azure Client objects and supporting services.
+// Where practical the original function signature is preserved. Client provides higher
+// level operations atop AzureProvider.
+type AzureProvider interface {
+	MSGraphClient
+}
```

Comment:
I wonder if we need the `AzureProvider` interface? I don't see the value it provides when it embeds `MSGraphClient` and only has a single implementation. I _think_ this would still compile if you removed these lines?